normalize.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"bytes"
+	"crypto/md5"
+	"reflect"
+	"sync"
+	"unicode"
+	"unsafe"
+)
+
+// Digest generates a digest(or sql-id) for a SQL.
+// the purpose of digest is to identity a group of similar SQLs, then we can do other logic base on it.
+func Digest(sql string) string {
+	d := digesterPool.Get().(*digester)
```

Comment:
how about:
```go
d := digesterPool.Get().(*digester)
defer func() {
	d.buffer.Reset()
	digesterPool.Put(d)
}() 
```

and remove line 48~49?

Reply:
if we should better don't use `defer` as we can in hot-code path and I have do some refactor PTAL :D 
https://medium.com/i0exception/runtime-overhead-of-using-defer-in-go-7140d5c40e32